bcipy/helpers/raw_data.py

```diff
@@ -58,22 +62,63 @@ def channels(self) -> List[str]:
     @property
     def numeric_data(self) -> pd.DataFrame:
         """Data for columns with numeric data. This is usually comprised of the
-        timestamp column and device channels, excluding triggers."""
+        timestamp column and device channels, excluding string triggers."""
         return self.dataframe.select_dtypes(exclude=['object'])
 
-    def by_channel(self) -> np.ndarray:
+    def by_channel(self, transform: Optional[Composition] = None) -> np.ndarray:
         """Data organized by channel.
 
+        Optionally, it can apply a BciPy Composition to the data before returning using the transform arg.
+        This will apply n tranformations to the data before returning. For an example Composition with
+        EEG preprocessing, see bcipy.signal.get_default_transform().
+
         Returns
         ----------
-        C x N numpy array with samples where C is the number of channels and N
-        is number of time samples."""
+        data: C x N numpy array with samples where C is the number of channels and N
+        is number of time samples
+        fs: resulting sample rate if any transformations applied"""
 
         numeric_data = self.numeric_data
 
         numeric_vals = numeric_data.values
         numeric_column_count = numeric_vals.shape[1]
-        return numeric_vals[:, 1:numeric_column_count].transpose()
+        data = numeric_vals[:, 1:numeric_column_count].transpose()
+        fs = self.sample_rate
+
+        if transform:
```

Comment:
It seems alright to filter out other Falsey things in this check ?  It could be useful to assert it is a Composition here more explicitly if you would want to see it more sensitive to the inputs expected.